src/applications/vaos/containers/VAFacilityPage.jsx

```diff
@@ -41,14 +42,43 @@ const uiSchema = {
     'ui:title':
       'Appointments are available at the following locations. Some types of care are only available at one location. Select your preferred location',
     'ui:widget': 'radio',
+    'ui:validations': [
+      (errors, vaFacility, data) => {
+        if (vaFacility && !vaFacility.startsWith(data.vaSystem)) {
+          errors.add(
+            'Please choose a facility that is in the selected VA health systems',
+          );
+        }
+      },
+    ],
     'ui:options': {
       hideIf: data => !data.vaSystem,
     },
   },
+  vaFacilityLoading: {
```

Comment:
In order to display messages in between the VA systems options and the submit buttons, I'm sometimes putting extra fields in the schema, which will use the field components in here to display messages. It's very weird, but this is the best way I found of doing it.